setup.py

```diff
@@ -1,26 +1,35 @@
-import setuptools
+from setuptools import setup
 
-setuptools.setup(
+setup(
     name="ramanujan",
     version="0.0.1",
+    python_requires=">=3.10.8, <3.11", # as of now, numba (including llvmlite) and ortools do not support python3.11, see respectively:
+    # https://github.com/numba/numba/issues/8304 ; https://github.com/google/or-tools/issues/3515
     description="Ramanujan Machine",
     packages=['ramanujan'],
     install_requires=[
         'cycler>=0.10.0',
         'kiwisolver>=1.1.0',
-        'matplotlib>=3.2.0',
-        'mpmath>=1.1.0',
-        'numpy>=1.18.1',
+        'matplotlib>=3.3.3',
+        'mpmath>=1.2.1',
+        'numpy>=1.19.5',
         'ordered-set>=3.1.1',
+        'ortools>=7.4.7247',
         'pandas>=1.0.1',
         'protobuf>=3.11.3',
+        'psutil>=5.5.1',
+        'psycopg2>=2.8.6',
+        'pybloom-live',
         'PyLaTeX>=1.3.1',
         'pyparsing>=2.4.6',
+        'pytest>=6.2.4',
         'python-dateutil>=2.8.1',
         'pytz>=2019.3',
+        'scipy>=1.6.0',
+        'simplejson>=3.16.0',
         'six>=1.14.0',
+        'sqlacodegen>=2.3.0',
         'sympy>=1.5.1',
-        'ortools>=7.4.7247',
-        'pybloom-live'
+        'xlrd>=2.0.1',
```

Comment:
Do all people who use all code in Ramanujan need all of these dependencies?
Perhaps we can `separate` the dependencies into subprojects, so if someone only needs one part they don't need to have everything;

In terms of easily installing everything and running something in Docker or some environment, this is probably the best, at least for now;

Leaving this here to think about solutions similar to yarn workspaces or cargo subpackages etc.

Reply:
All I did was add the dependencies from the other repo. I haven't investigated which dependencies I actually need (and chances are I need most), but if we really want to minimize dependencies I think the best call would be to ask everyone to investigate which dependencies their projects actually need. (otherwise we can have just one guy combing through the code, but that's probably going to involve unnecessary headache...)
If subproject-specific dependencies are possible and easy enough to work with, I think that could be a good idea.